waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -614,28 +612,33 @@ proc validateRoot*(wakuRlnRelay: WakuRLNRelay, root: MerkleNode): bool =
   ## Validate against the window of roots stored in wakuRlnRelay.validMerkleRoots
   return root in wakuRlnRelay.validMerkleRoots
 
-proc toMembershipKeyPairs*(groupKeys: seq[(string, string)]): seq[
-    MembershipKeyPair] {.raises: [Defect, ValueError].} =
+proc toMembershipKeyPairs*(groupKeys: seq[(string, string)]): RlnRelayResult[seq[
+    MembershipKeyPair]] =
   ## groupKeys is  sequence of membership key tuples in the form of (identity key, identity commitment) all in the hexadecimal format
   ## the toMembershipKeyPairs proc populates a sequence of MembershipKeyPairs using the supplied groupKeys
 
   var groupKeyPairs = newSeq[MembershipKeyPair]()
 
   for i in 0..groupKeys.len-1:
-    let
-      idKey = hexToUint[IDKey.len*8](groupKeys[i][0]).toBytesLE()
-      idCommitment =  hexToUint[IDCommitment.len*8](groupKeys[i][1]).toBytesLE()
-    groupKeyPairs.add(MembershipKeyPair(idKey: idKey,
-        idCommitment: idCommitment))
-  return groupKeyPairs
-
-proc calcMerkleRoot*(list: seq[IDCommitment]): string {.raises: [Defect, IOError].} =
+    try:
+      let
+        idKey = hexToUint[IDKey.len*8](groupKeys[i][0]).toBytesLE()
+        idCommitment = hexToUint[IDCommitment.len*8](groupKeys[i][1]).toBytesLE()
+      groupKeyPairs.add(MembershipKeyPair(idKey: idKey,
+          idCommitment: idCommitment))
+    except ValueError as err:
+      warn "could not convert the group key to bytes", err = err.msg
+      return err("could not convert the group key to bytes: " & err.msg)
+  return ok(groupKeyPairs)
+
+proc calcMerkleRoot*(list: seq[IDCommitment]): RlnRelayResult[string] =
   ## returns the root of the Merkle tree that is computed from the supplied list
```

Comment:
The proc description should change to indicate that the Result output may contain an error if the operation does not go through. This shall be done for all the procs in this PR.

Reply:
Addressed in [3570dd1](https://github.com/status-im/nwaku/pull/1321/commits/3570dd18f9dbf444614b7eb5b767e0b11ce2a4da)